### PR TITLE
[posix] remove use of core-internal types in `netif.cpp` platform

### PR DESCRIPTION
**Pull request by abtink**
---

Removes the use of `ot::Ip6::Address` and `OT_ASSERT()` in posix `netif.cpp` module (remove dependency to `"net/ip6_address.hpp"` and `"common/debug.hpp"` headers and their definitions).

### CODE REVIEW

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -340,31 +339,55 @@ static void LogDebg(const char *aFormat, ...)
...
 #if defined(__APPLE__) || defined(__NetBSD__) || defined(__FreeBSD__)
-static const uint8_t allOnes[] = {0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
-                                  0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF};
+
+static const uint8_t kAllOnes[] = {
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+};
 
 #define BITS_PER_BYTE 8
 #define MAX_PREFIX_LENGTH (OT_IP6_ADDRESS_SIZE * BITS_PER_BYTE)
```
Comment (superwhd) [inline src/posix/platform/netif.cpp:348]: nit: make them constants? Maybe in a future PR.

**Review by superwhd — APPROVED** [review]
LGTM 👍

File: src/posix/platform/netif.cpp (hunk 1 above)
Comment (abtink) [reply src/posix/platform/netif.cpp:348]: Yes. Will leave for future PR. We can  address these in smaller PRs over time.
There are actually more cases similar to this (constants in `posix` platform modules where `#define` or `enum` is used and we can change to `constexpr`).

**Review by jwhui — APPROVED** [review]

**Pull request merged**